src/qibolab/_core/instruments/qblox/config.py

```diff
@@ -0,0 +1,153 @@
+            str(self.ports[0])
+            if self.ports[1] is None
+            else f"{self.ports[0]}_{self.ports[1]}"
+        )
+        return f"{direction}{channels}"
+
+
+def _probe(id_: ChannelId, channel: Channel) -> Optional[ChannelId]:
+    return (
+        id_
+        if isinstance(channel, IqChannel)
+        else (
+            channel.probe
+            if isinstance(channel, AcquisitionChannel) and channel.probe is not None
+            else None
+        )
+    )
+
+
+def module(
+    mod: Module,
+    mod_channels: dict[ChannelId, PortAddress],
+    channels: dict[ChannelId, Channel],
+    configs: Configs,
+):
+    # map sequencers to specific outputs (but first disable all sequencer connections)
+    mod.disconnect_outputs()
+
+    if mod.is_qrm_type:
+        # we do not currently support acquisition on external digital trigger
+        mod.scope_acq_trigger_mode_path0("sequencer")
+        mod.scope_acq_trigger_mode_path1("sequencer")
+
+    # set lo frequencies
+    los = {
+        (probe, cast(IqChannel, channels[probe]).lo)
+        for probe in (_probe(ch, channels[ch]) for ch in mod_channels)
+        if probe is not None
+    }
+    for probe, lo in los:
+        if lo is None:
+            continue
+        n = mod_channels[probe].ports[0]  # TODO: check it is the correct path
+        getattr(mod, f"out{n}_lo_freq")(cast(OscillatorConfig, configs[lo]).frequency)
+
+
+def sequencer(
+    seq: Sequencer,
+    address: PortAddress,
+    sequence: Q1Sequence,
+    channel_id: ChannelId,
+    channels: dict[ChannelId, Channel],
+    configs: Configs,
+    acquisition: AcquisitionType,
+):
+    # upload sequence
+    # - ensure JSON compatibility of the sent dictionary
+    seq.sequence(json.loads(sequence.model_dump_json()))
+
+    # configure the sequencers to synchronize
+    seq.sync_en(True)
+
+    config = configs[channel_id]
+
+    # set parameters
+    # offsets
+    seq.offset_awg_path0(config.offset if isinstance(config, DcConfig) else 0.0)
+    seq.offset_awg_path1(0.0)
+    # modulation, only disable for QCM - always used for flux pulses
+    mod = cast(Module, seq.ancestors[1])
+    seq.mod_en_awg(mod.is_qrm_type or mod.is_rf_type)
+    # acquisition
+    if address.input:
+        assert isinstance(config, AcquisitionConfig)
+        seq.integration_length_acq(1000)
+        # discrimination
+        seq.thresholded_acq_rotation(config.iq_angle)
+        seq.thresholded_acq_threshold(config.threshold)
+        # demodulation
+        seq.demod_en_acq(acquisition is not AcquisitionType.RAW)
+
+    probe = _probe(channel_id, channels[channel_id])
+    if probe is not None:
+        freq = cast(IqConfig, configs[probe]).frequency
+        lo = cast(IqChannel, channels[probe]).lo
+        assert lo is not None
+        lo_freq = cast(OscillatorConfig, configs[lo]).frequency
+        seq.nco_freq(freq - lo_freq)
+
+    # connect to physical address
+    seq.connect_sequencer(address.local_address)
```

Comment:
This line is giving the following error when running on iqm5q:
```py
ValueError: connection command 'out4' (index 0): dac is out of range: 4
```
I guess it is not related to this line, but some other setting that comes before that. It appears for channel `3/flux` which is connected to `iqm5q_qblox_module2, slot=2, ports=(4, None)`. The offset in the runcard is <1, I am not sure which value is out of range.

Reply:
Actually, the line setting it is this one, which is pretty explicit
https://github.com/qiboteam/qibolab_platforms_qrc/blob/98d7158238384d19f16655e3d038166421c0d5a2/iqm5q/platform.py#L23
but I've taken it from the original 0.1 platform, which is also explicit
https://github.com/qiboteam/qibolab_platforms_qrc/blob/c0a92c97684fe627628bc25fee7a936e0fa08efd/iqm5q/platform.py#L74-L77

So, I should check how that is used in the 0.1 driver, and translated to Qblox layout. But I do not have any simple explanation (yet).